rbac/management/notifications/notification_hanlders.py

```diff
@@ -0,0 +1,179 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Notification handlers of object change."""
+import logging
+
+from django.conf import settings
+from management.notifications.producer_util import NotificationProducer
+
+from api.models import Tenant
+
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+producer = NotificationProducer()
+
+
+def notify_all(event_type, payload):
+    """Notify all tenants."""
+    # To avoid memory overloaded, use iterator:
+    # https://docs.djangoproject.com/en/4.0/ref/models/querysets/#django.db.models.query.QuerySet.iterator
+    for tenant in Tenant.objects.exclude(tenant_name="public").iterator():
+        # Tenant name pattern is acct12345
+        account_id = tenant.tenant_name.split("acct")[-1]
+        producer.send_kafka_message(event_type, account_id, payload)
+
+
+def handle_system_role_change_notification(role_obj, operation):
+    """Signal handler for sending notification message when system Role object changes."""
+    if not settings.NOTIFICATION_ENABLED:
+        return
+
+    payload = {"username": "Red Hat", "name": role_obj.name, "uuid": str(role_obj.uuid)}
```

Comment:
Would it make sense to have some sort of payload factory we could use to DRY up some of the repetition in building these notifications?

Reply:
Yea, good idea, let me try